aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -106,13 +124,38 @@ pub fn get_snapshot_field_by_type(
     }
 }
 
+pub fn get_snapshot_field_with_native_value(
+    ty_arg: &Type,
+    agg_snapshot: &StructRef,
+) -> SafeNativeResult<DelayedFieldID> {
+    match ty_arg {
+        Type::U128 | Type::U64 => {
+            let value = get_struct_field(agg_snapshot, AGG_SNAPSHOT_VALUE_FIELD_INDEX)?
+                .value_as::<SizedID>()?;
+            Ok(value.into())
+        },
+        _ => Err(SafeNativeError::Abort {
+            abort_code: EUNSUPPORTED_AGGREGATOR_SNAPSHOT_TYPE,
+        }),
+    }
+}
+
 pub fn get_derived_string_field(derived_string: &StructRef) -> SafeNativeResult<Vec<u8>> {
     Ok(string_to_bytes(
         get_struct_field(derived_string, DERIVED_STRING_VALUE_FIELD_INDEX)?.value_as::<Struct>()?,
     )
     .map_err(PartialVMError::from)?)
 }
 
+pub fn get_derived_string_field_with_native_value(
+    derived_string: &StructRef,
+) -> SafeNativeResult<Vec<u8>> {
+    Ok(string_to_bytes(
+        get_struct_field(derived_string, DERIVED_STRING_VALUE_FIELD_INDEX)?.value_as::<Struct>()?,
+    )
```

Comment:
Good catch! Haven't even noticed because ended up not using this function, `native_read_derived_string` has the correct impl: 
```
let id: DelayedFieldID = safely_pop_arg!(args, SizedID).into();
        resolver
            .validate_delayed_field_id(&id)
            .map_err(PartialVMError::from)?;
```